src/core/noteLocationBySubject.ts

```diff
@@ -0,0 +1,232 @@
+// This regexs from https://github.com/regexhq/mentions-regex/blob/master/index.js
```

Comment:
some comments here would be nice, it's very hard to understand what is going on here

Reply:
I added some comments. In addition, I will post in the forum about how this mechanism works but not now.